packages/api/cli/src/util/check-system.ts

```diff
@@ -27,43 +27,47 @@ async function checkNodeVersion() {
   return process.versions.node;
 }
 
-const NPM_ALLOWLISTED_VERSIONS = {
-  all: '^3.0.0 || ^4.0.0 || ~5.1.0 || ~5.2.0 || >= 5.4.2',
-  darwin: '>= 5.4.0',
-  linux: '>= 5.4.0',
-};
-const YARN_ALLOWLISTED_VERSIONS = {
-  all: '>= 1.0.0',
+async function checkPnpmNodeLinker() {
+  const nodeLinkerValue = await spawnPackageManager(['config', 'get', 'node-linker']);
+
+  if (nodeLinkerValue !== 'hoisted') {
+    throw new Error('When using pnpm, `node-linker` must be set to "hoisted". Run `pnpm config set node-linker hoisted` to set this config value.');
+  }
+}
+
+const ALLOWLISTED_VERSIONS: Record<SupportedPackageManager, Record<string, string>> = {
+  npm: {
+    all: '^3.0.0 || ^4.0.0 || ~5.1.0 || ~5.2.0 || >= 5.4.2',
+    darwin: '>= 5.4.0',
+    linux: '>= 5.4.0',
+  },
+  yarn: {
+    all: '>= 1.0.0',
+  },
+  pnpm: {
+    all: '>= 8.0.0',
```

Comment:
I don't know if this is necessary, but the Node compatibility matrix in the pnpm docs only lists up to pnpm 8 so I feel like this is a decent lower bound: https://pnpm.io/installation#compatibility